File: go/order.go
package zymeup

type OrderClient struct {
	client *Client
}

func NewOrderClient(c *Client) *OrderClient {
	return &OrderClient{client: c}
}

func (c *OrderClient) List(params map[string]interface{}) ([]map[string]interface{}, error) {
	path := "/api/v1/order/list" + buildQueryString(params)
	resp, err := c.client.doRequest("GET", path, nil)
	if err != nil {
		return nil, err
	}
	var result struct {
		Code int `json:"code"`
		Data struct {
			Items []map[string]interface{} `json:"items"`
			Total int                      `json:"total"`
		} `json:"data"`
	}
	if err := decodeResponse(resp, &result); err != nil {
		return nil, err
	}
	return result.Data.Items, nil
}

func (c *OrderClient) Get(id string) (map[string]interface{}, error) {
	resp, err := c.client.doRequest("GET", "/api/v1/order/"+id, nil)
	if err != nil {
		return nil, err
	}
	var result struct {
		Code int                    `json:"code"`
		Data map[string]interface{} `json:"data"`
	}
	if err := decodeResponse(resp, &result); err != nil {
		return nil, err
	}
	return result.Data, nil
}

func (c *OrderClient) Create(data map[string]interface{}) (map[string]interface{}, error) {
	resp, err := c.client.doRequest("POST", "/api/v1/order/create", data)
	if err != nil {
		return nil, err
	}
	var result struct {
		Code int                    `json:"code"`
		Data map[string]interface{} `json:"data"`
	}
	if err := decodeResponse(resp, &result); err != nil {
		return nil, err
	}
	return result.Data, nil
}

func (c *OrderClient) Update(id string, data map[string]interface{}) (map[string]interface{}, error) {
	resp, err := c.client.doRequest("POST", "/api/v1/order/"+id+"/update", data)
	if err != nil {
		return nil, err
	}
	var result struct {
		Code int                    `json:"code"`
		Data map[string]interface{} `json:"data"`
	}
	if err := decodeResponse(resp, &result); err != nil {
		return nil, err
	}
	return result.Data, nil
}

func (c *OrderClient) Cancel(id string) error {
	resp, err := c.client.doRequest("POST", "/api/v1/order/"+id+"/cancel", nil)
	if err != nil {
		return err
	}
	return decodeResponse(resp, nil)
}

func (c *OrderClient) CreateWithDocuments(data map[string]interface{}) (map[string]interface{}, error) {
	resp, err := c.client.doRequest("POST", "/api/v1/order/create-with-documents", data)
	if err != nil {
		return nil, err
	}
	var result struct {
		Code int                    `json:"code"`
		Data map[string]interface{} `json:"data"`
	}
	if err := decodeResponse(resp, &result); err != nil {
		return nil, err
	}
	return result.Data, nil
}

File: python/zymeup_sdk/order.py
"""Zymeup Python SDK - Order Client"""
from typing import Optional, Dict, Any, List


class OrderClient:
    """Order management client"""

    def __init__(self, client):
        self._client = client

    def list(self, page: int = 1, page_size: int = 25, status: Optional[str] = None) -> Dict[str, Any]:
        """List orders"""
        params = {"page": page, "page_size": page_size}
        if status:
            params["status"] = status
        return self._client.request("GET", "/api/v1/order/list", params=params)

    def get(self, order_id: str) -> Dict[str, Any]:
        """Get order detail"""
        return self._client.request("GET", f"/api/v1/order/{order_id}")

    def create(self, data: Dict[str, Any]) -> Dict[str, Any]:
        """Create order"""
        return self._client.request("POST", "/api/v1/order/create", json=data)

    def create_with_documents(self, data: Dict[str, Any]) -> Dict[str, Any]:
        """Create order with EPOD/ECMR documents"""
        return self._client.request("POST", "/api/v1/order/create-with-documents", json=data)

    def update(self, order_id: str, data: Dict[str, Any]) -> Dict[str, Any]:
        """Update order"""
        return self._client.request("POST", f"/api/v1/order/{order_id}/update", json=data)

    def cancel(self, order_id: str) -> Dict[str, Any]:
        """Cancel order"""
        return self._client.request("POST", f"/api/v1/order/{order_id}/cancel")
